src/components/Accordion/types.ts

```diff
@@ -17,43 +17,30 @@ export type AriaAccordionItemProps = {
     header: Omit<AccordionHeaderProps, 'isOpen'> & { active?: boolean; onClick?: () => void };
     padding?: boolean;
     divider?: boolean;
-    headerComponent?: FC<AccordionHeaderProps>;
+    headerComponent?: (props: AccordionHeaderProps) => JSX.Element | null;
```

Comment:
```suggestion
    headerComponent?: (props: AccordionHeaderProps) => ReactNode | null;
```
So you can have `ReactElement` inside and even `Nullable<ReactElement>`

Reply:
With JSX.Element too, or am I wrong? I will change it and also bring back the type you added on Clarify for the `Header` component

Reply:
I was researching about this and `JSX.Element` would be the best solution here since `ReactNode` represents anything that `React` can render, including `string`, `boolean`, `number` and so on... But we must provide a component as the `headerComponent` prop and `React` will type it like this:
```
declare global {
  namespace JSX {
    interface Element extends React.ReactElement<any, any> { }
  }
}
```

Reply:
@imoutaharik and I were talking about this and we believe the way to go is using ReactElement also to keep things consistent in Fondue since we use it in other places.

Reply:
ReactElement doesn't allow TextNode, where ReactNode does.

Reply:
You can go for it if you feel like 👍

Reply:
Went for it since it was already used on Jonas PR too